buildpacks/ruby/src/test_helper.rs

```diff
@@ -0,0 +1,80 @@
+use crate::RubyBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::data::buildpack_plan::Entry;
+use libcnb::data::stack_id;
+use libcnb::data::{buildpack::SingleBuildpackDescriptor, buildpack_plan::BuildpackPlan};
+use libcnb::detect::DetectContext;
+use libcnb::generic::{GenericMetadata, GenericPlatform};
+use libcnb::Platform;
+use std::fs;
+use std::path::PathBuf;
+
+pub fn touch_file(path: &PathBuf, f: impl FnOnce(&PathBuf)) {
+    if let Some(parent) = path.parent() {
+        println!("path {:?}", path);
+        println!("parent {:?}", parent);
+        if !parent.exists() {
+            std::fs::create_dir_all(parent).unwrap();
+        }
+    }
+    std::fs::write(path, "").unwrap();
+    f(path);
+    std::fs::remove_file(path).unwrap();
+}
+
+#[allow(dead_code)]
```

Comment:
There are 15 usages of `#[allow(dead_code)]` in the repo. I'm presuming some are redundant (and the code is now being used, but the annotation is leftover), and maybe some is genuinely deadcode? It feels like it would be good to remove all of these annotations, along with anything that's actually deadcode :-)

Reply:
This review comment isn't resolved

Reply:
I've removed all the dead_code I could find